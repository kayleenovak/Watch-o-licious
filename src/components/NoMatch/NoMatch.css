.four-o-four {
  width: 100%;
  text-align: center;
}

.error-page {
  color: white;
}