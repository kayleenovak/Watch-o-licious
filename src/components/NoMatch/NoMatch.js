import React from 'react'
import { NavLink } from 'react-router-dom'
import './NoMatch.css'

export const NoMatch = () => {
  return (
    <section className='four-o-four'>
      <h3 className='error-page'>404 Error</h3>
      <NavLink to='/'>Click here to return Home</NavLink>
    </section>
    )
}