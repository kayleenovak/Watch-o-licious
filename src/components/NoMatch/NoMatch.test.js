import React from 'react'
import { shallow } from 'enzyme'
import { NoMatch } from './NoMatch'
import { NavLink } from 'react-router-dom'

describe('NoMatch', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<NoMatch />)

    expect(wrapper).toMatchSnapshot()
  })
  it('should return a Redirect', () => {
    const wrapper = shallow(<NoMatch />)

    expect(wrapper.find(NavLink).length).toBe(1)
    expect(wrapper.find('h3').length).toBe(1)
    expect(wrapper.find('section').length).toBe(1)
  })
})