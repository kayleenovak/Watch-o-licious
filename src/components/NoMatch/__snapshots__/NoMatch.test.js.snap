// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NoMatch />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="error-page"
        >
          404 Error
        </h3>,
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        >
          Click here to return Home
        </NavLink>,
      ],
      "className": "four-o-four",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "404 Error",
          "className": "error-page",
        },
        "ref": null,
        "rendered": "404 Error",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": "Click here to return Home",
          "to": "/",
        },
        "ref": null,
        "rendered": "Click here to return Home",
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="error-page"
          >
            404 Error
          </h3>,
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/"
          >
            Click here to return Home
          </NavLink>,
        ],
        "className": "four-o-four",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "404 Error",
            "className": "error-page",
          },
          "ref": null,
          "rendered": "404 Error",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Click here to return Home",
            "to": "/",
          },
          "ref": null,
          "rendered": "Click here to return Home",
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
